### DISCUSSION TITLE
json format after 2.3.x

### DISCUSSION THREAD

**Discussion by keyiflerolsun** [pallets/flask#5095] [Q&A] [answered]
how can i do this operation after 2.3.x

```python
app.config["JSON_SORT_KEYS"]              = False
app.config["JSONIFY_PRETTYPRINT_REGULAR"] = True
app.config["JSON_AS_ASCII"]               = False
```

**Comment by davidism** [comment, marked as answer]
Described in the API docs, they have moved to attributes on the default provider class: https://flask.palletsprojects.com/en/2.3.x/api/?highlight=sort_keys#flask.json.provider.DefaultJSONProvider. Also, using these raised deprecation warnings in 2.2.x, be sure to run your tests with deprecation warnings treated as errors so that you get notified of those types of changes early.

**Reply by keyiflerolsun** [reply]
```python
app = Flask(__name__)
app.json_provider_class.ensure_ascii = False
app.json_provider_class.sort_keys    = False
app.json_provider_class.compact      = False
```
